packages/api-server/api_server/routes/fleets.py

```diff
@@ -83,8 +83,9 @@ async def get_robots(
                 TaskStateEnum.QUEUED.value,
             ]
             tasks = await ttm.TaskSummary.filter(
-                state__in=filter_states, **filter_params
-            )
+                state__in=filter_states,
+                **filter_params,
+            ).order_by("start_time")
```

Comment:
> position: 1 | task A: start time: 15:00 | endtime: None (still in progress)
> position: 2 | task B: start time: 15:02 | endtime: 15:10

I think in such a case, another robot will be handling task B. From what I observe so far, for one robot, a task will always have an estimated end time and the robot does not start on a new task until the task it is currently executing ends. I may be wrong about how rmf does the task assignment though and there may be more complicated scenarios that I have not encountered. If it goes over the estimated end time, then it displays `delayed active`.

> IMO, I think that when a task finishes it loses priority, so task B at 15:11 should be above task A. Maybe we handle that just in the UI (in the future).

The current behavior is such that when a task has finished, it will not be included in the task list as it has been filtered out. I think priority here refers to the number that we set when we create task. When we sort by `'-priority,-start_time'`, the task assigned with highest priority number will almost always be the first task and it may not be the currently active task. When the user clicks on the robot, the robot info panel would not show the currently executing task as a result.